### ISSUE TITLE
How to add auxiliary decoder head for Swin Transformer?

### ISSUE THREAD

**Issue by edizhuang** [open-mmlab/mmsegmentation#1884]
Hi,

1. How to add auxiliary decoder head for Swin Transformer? And how to set it in config file (in '_base_' and 'swin')? 

By default upernet_swin_tiny config file already has an auxiliary decoder head of FCN, so adding another there will be two (I see function _init_auxiliary_head() takes input of list, but don't know the right way to define in config files).

2. And by the way, is it possible to add auxiliary decoder head for parallel task (e.g. edge prediction) with corresponding ground truth to help semantic segmentation prediction? Just don't know how to do that in mmsegmentation.

Thanks,

**Comment by MengzhangLI** [comment]
1. Check other config files which has auxiliary heads settings such as [this one](https://github.com/open-mmlab/mmsegmentation/blob/master/configs/_base_/models/bisenetv1_r18-d32.py#L40-L65). Use `in_index` to determine which feature map from backbone as input for certain auxiliary head. You can debug on your own if I do not explain clearly.

2. You can read [STDC](https://github.com/open-mmlab/mmsegmentation/tree/master/configs/stdc), it uses boundary information to calculate loss. Code is [here](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/models/decode_heads/stdc_head.py#L9-L85). MMSegmentation do not have other parallel task/ post-processing/ novel loss right now, if you want to add it, PR is welcome.

**Comment by edizhuang** [comment]
Hi,

If I have two inputs separately to forward and get results, how could I combine the losses?

I see in losses are generated in function forward_train in decode_head.py, 
        seg_logits = self.forward(inputs)
        losses = self.losses(seg_logits, gt_semantic_seg)
        return losses

And losses function is here:
    @force_fp32(apply_to=('seg_logit', ))
    def losses(self, seg_logit, seg_label):
        """Compute segmentation loss."""
        loss = dict()
        seg_logit = resize(
            input=seg_logit,
            size=seg_label.shape[2:],
            mode='bilinear',
            align_corners=self.align_corners)
        if self.sampler is not None:
            seg_weight = self.sampler.sample(seg_logit, seg_label)
        else:
            seg_weight = None
        seg_label = seg_label.squeeze(1)
        loss['loss_seg'] = self.loss_decode(
            seg_logit,
            seg_label,
            weight=seg_weight,
            ignore_index=self.ignore_index)
        loss['acc_seg'] = accuracy(seg_logit, seg_label)
        return loss

Thanks,
